### PR TITLE
complete and total revert of adam changes to tinfoil hat code

### LINKED ISSUE #34327: Agent ID card removes Sec HuD
(WEB REPORT BY: calzilla1 REMOTE: 206.221.180.138:7777)
# Revision
6dde500ce254d492226844f7b578dfdcf87511f4

# Description
>Be flukie
>Want to SLETH it
>Change name to Kevin Smith, head to locker room and change clothes, jumpsuit, bag, everything. No witnesses
>Nukies go loud, head to the bridge to check for an opening
>GET IMMEDIATELY LIT UP BY SEC
>Die
>deadchat salt incident of 2023
>Seethe at the homosexual security birds on why they lit up some seemingly random assistant
>No HuD

This is gay and retarded and I demand the choder who broke this to be valided by shitcurity for a thousand years
# Steps to Reproduce
Buy agent ID card
Set a name
Have gay bird sec see you
Cry and die
# What you Expected
For my character to have a HuD
# What Actually Happened
No HuD and no nuke disk

### PR DESCRIPTION
**Pull request by nervere**
## What this does
Completely reverts all gameplay and changes in Adam's tinfoil hat PR: #32664
Fixes #34327

## Why it's good
- The PR was chock-full of sneaky-coded changes to balance that were not in the changelog.
- The PR completely misrepresented how its changed would affect gameplay.
- The PR fundamentally changed the behavior of many classic items to functionality that does not make sense.
- The PR broke agent ID cards, HUDs (if you were wearing completely arbitrary items like holding a megaphone???), telepathy, and probably a lot more that I don't care to bugtest!
- _**Slightly better code readability is not worth the complete butchering of the way our game's items function.**_

## Changelog

:cl:
 * bugfix: Agent ID cards should now properly display HUD information.
 * bugfix: Voice changers no longer magically block telepathy, but do still block AI tracking.
 * bugfix: Stun helmets no longer block AI tracking, but still block telepathy and other electrical infetterence.
 * bugfix: Reverted stealth-buff to full-power vampires that let them block all tracking / HUD / etc.
 * tweak: Genetics: "Jam Signals Block" restored to "Psy Resist Block"

### CODE REVIEW

**Comment by Exxion** [comment]
What's that map?

**Comment by nervere** [comment]
It's the grey mothership research lab. I have no idea why it didn't like the "VOX_BLOOD" define. I think the only reason the map bot was looking at it is because I changed one of the gene injectors that Adam changed within it.

I have a bunch of changed lines on the map because, when I re-saved it to fix it, it re-wrote the map in TGM format, which has a lot more lines. It would seem the original map creator made it in DMM format.

File: code/datums/gamemode/powers/changeling.dm [hunk 1]
```diff
@@ -226,6 +226,7 @@
 		return
 	var/mob/living/carbon/human/C = R.antag.current
 	to_chat(C, "<span class='notice'>We distort our form to prevent AI-tracking.</span>")
+	C.digitalcamo = 1
```
Comment (pedr69) [inline code/datums/gamemode/powers/changeling.dm:229]: hang on the digital camo didn't even work? hahahahaha

Comment (nervere) [reply code/datums/gamemode/powers/changeling.dm:229]: He removed digital camo and put it under a single giant proc that was meant to handle every single kind of anti-tracking, which went about as well as you'd expect. I re-added the digital camo variable and undid his changes to the proc.

**Comment by Exxion** [comment]
Adam wanted me to mention that the voice changer and mad scientist megaphone already blocked AI tracking before his PR, and still do after this one. He also wanted me to mention something about vampires, but I'm not going to because his reasoning was stupid
He wasn't clear on why he couldn't tell you this himself

**Comment by nervere** [comment]
I appreciate you relaying the message, but this is just more of Adam omitting the whole truth about his PR's. I won't subject you to a debate about his awful code changes with you as the messenger, but I think him being repo-banned for this very same behavior more than speaks for itself.

**Comment by Exxion** [comment]
I mean the voice changer thing is relevant because your changelog says it no longer blocks AI tracking, which is not true

**Comment by nervere** [comment]
I see, you're right. I've adjusted the changelog accordingly.

**Comment by Kurfursten** [comment]
I think you got the stun helmet completely backwards. The Stun Helmet is supposed to block all mind-based powers like telepathy but NOT block AI tracking.

`desc = "For the experimental program of deploying armless security officers. Its complex wiring is known to block out psychic powers and 5G signals."`

**Comment by Eneocho** [comment]
> I think you got the stun helmet completely backwards. The Stun Helmet is supposed to block all mind-based powers like telepathy but NOT block AI tracking.
> 
> `desc = "For the experimental program of deploying armless security officers. Its complex wiring is known to block out psychic powers and 5G signals."`

But what if the AI tracks you via 5G?

**Comment by Servalfang** [comment]
> But what if the AI tracks you via 5G?

Stimulate your sensors.

File: code/datums/gamemode/powers/changeling.dm (hunk 1 above)
Comment (Inorien) [reply code/datums/gamemode/powers/changeling.dm:229]: image

**Comment by nervere** [comment]
> I think you got the stun helmet completely backwards. The Stun Helmet is supposed to block all mind-based powers like telepathy but NOT block AI tracking.
> 
> `desc = "For the experimental program of deploying armless security officers. Its complex wiring is known to block out psychic powers and 5G signals."`

My apologies, you're right. I have it correct in the code, but I messed up the changelog again.
I have adjusted the changelog to reflect what you said.

File: code/game/dna/genes/goon_powers.dm [hunk 2]
```diff
@@ -614,7 +612,7 @@
 		to_chat(user, "<span class='warning'>This can only be used on carbon beings.</span>")
 		return 1
 
-	if (!can_mind_interact(M.mind))
+	if (M_PSY_RESIST in M.mutations)
 		to_chat(user, "<span class='warning'>You can't see into [M.name]'s mind at all!</span>")
```
Comment (pedr69) [inline code/game/dna/genes/goon_powers.dm:616]: how come tinfoil blocks telepathic messages but doesn't stop the government from READING YOUR MIND and taking your bank PIN

Comment (Kurfursten) [reply code/game/dna/genes/goon_powers.dm:616]: tinfoil is intended to block all mind powers

Comment (pedr69) [reply code/game/dna/genes/goon_powers.dm:616]: wouldnt it neater and overall less retarded if putting on a tinfoil hat just gave you PSY_RESIST?

**Comment by pedr69** [comment]
bro... you fix the tinfoil empathy...

**Pull request merged**